### DISCUSSION TITLE
Application logs to the console

### DISCUSSION THREAD

**Discussion by martimors** [nginx/unit#1494] [Support] [answered]
Hi team, thank you for an interesting FOSS product.

I have a simple web-app that has three components:
- UI (static Vue SPA)
- Backend (FastAPI app)
- Worker (Celery app)

Config as follows:

```json
{
  "listeners": {
    "*:80": {
      "pass": "routes"
    }
  },

  "routes": [
    {
      "match": {
        "uri": ["/api*"]
      },
      "action": {
        "pass": "applications/fastapi"
      }
    },
    {
      "match": {
        "uri": "/*"
      },
      "action": {
        "share": "/www/static$uri"
      }
    }
  ],
  "applications": {
    "fastapi": {
      "type": "python 3.12",
      "path": "/var/app",
      "module": "myapp.api",
      "callable": "app"
    },
    "worker": {
      "type": "python 3.12",
      "path": "/var/app",
      "module": "myapp.tasks",
      "callable": "run_celery_worker"
    }
  },
  "access_log": "/dev/stdout"
}

```

Dockerfile like this:

```Dockerfile
FROM node:22 AS build
WORKDIR /var/app
COPY . .
RUN npm run build

FROM unit:python3.12


WORKDIR /var/app
COPY --from=build /var/app/dist /www/static
RUN chown -R unit:unit .
COPY unit.config.json /docker-entrypoint.d/
COPY myapp .
COPY requirements.txt requirements.txt
COPY myapp myapp
RUN pip install -r requirements.txt

EXPOSE 8000
```

I can access the backend and frontend just fine in the browser, so the unit config seems to be correct.

Both the backend and the worker app is logging to the console, but when I run the application in unit, these logs are not seen anywhere. I do, however, see the unit access logs, but I would like to be able to also see the logs from my applications. Can this be achieved?

**Comment by ac000** [comment]
Perhaps by setting the applications `stdout` and or `stderr` [optons](https://unit.nginx.org/configuration/#applications).

**Reply by martimors** [reply]
Docs say those settings have no effect when running with the `--no-daemon` flag, which [the docker container does](https://github.com/nginx/unit/blob/master/pkg/docker/Dockerfile.python3.11#L89). It does say that `stdout` comes to the console and stderr comes to "units log", but I can't figure out how to configure this log in the container let alone pipe it to the console :/

**Reply by martimors** [reply]
Upon further inspection, the `/var/log/unit.log` file is symlinked to `/dev/stderr`, which confuses me further because then I would expect the logs to be visible by default. I can access the app through the browser so it is clearly running, and when running outside unit, logs are seen.

**Reply by ac000** [reply]
Maybe [this](https://github.com/nginx/unit-docs/issues/147#issuecomment-2127037950) will help...

**Comment by martimors** [comment, marked as answer]
I ended up ditching unit and deploying my backend and celery worker in a separate container, with an nginx container as a proxy in front. I do think the idea of unit is nice though, so would still like to understand a bit more whether my use of it in this example was somehow not aligned with the intended use-case.
